src/app/d2-vendors/VendorItemComponent.tsx

```diff
@@ -102,12 +106,16 @@ export default class VendorItemComponent extends React.Component<Props, {}> {
     } else {
       console.log(item);
 
-      const reviewData = trackerService.getD2ReviewDataCache().getRatingData(undefined, item.itemHash);
+      let reviewData: DimWorkingUserReview | null = null;
 
-      if (reviewData && !reviewData.reviews) {
-        const reviewsData = await trackerService.getItemReviewAsync(item.itemHash);
+      if (trackerService) {
+        reviewData = trackerService.getD2ReviewDataCache().getRatingData(undefined, item.itemHash);
 
-        Object.assign(reviewData, reviewsData);
+        if (reviewData && !reviewData.reviews) {
+          const reviewsData = await trackerService.getItemReviewAsync(item.itemHash);
+
+          Object.assign(reviewData, reviewsData);
```

Comment:
If we're mutating something here, we should call `setState` with the updated state in order to correctly re-render.